std/algorithm/iteration.d

```diff
@@ -4878,26 +4878,42 @@ private struct UniqResult(alias pred, Range)
 struct Permutations(Range)
     if (isRandomAccessRange!Range && hasLength!Range)
 {
-    size_t[] indices, state;
-    Range r;
+    private size_t[] _indices, _state;
+    private Range _r;
+    private bool _empty;
+
+    // Explicitly undocumented. It will be removed in December 2016. @@@DEPRECATED_2016-12@@@
```

Comment:
but you don't know what people do.That is the point here. Just because you wouldn't, does not mean somebody else wouldn't.